urllib3/connectionpool.py

```diff
@@ -790,11 +790,13 @@ def _prepare_proxy(self, conn):
             set_tunnel = conn.set_tunnel
         except AttributeError:  # Platform-specific: Python 2.6
             set_tunnel = conn._set_tunnel
-
+        host = self.host
+        if not (host.startswith('[') and host.endswith(']')):
+            host = '[' + host + ']'
```

Comment:
Ah, sorry. What about we add checks for `:` and not '.'. That should help with detecting for ipv6 strings

```
...
if not (host.startswith('[') and host.endswith(']')) and ':' in host and '.' not in host:
     ...
```

This could be moved as a helper function as well. 

Regarding to moving this to a higher level. I was not sure where to put this. The brackets was passed down correctly but stripped in the constructor to prevent double bracket later - https://github.com/shazow/urllib3/blob/master/urllib3/connectionpool.py#L71

Can you point to there it's appropriate to add the brackets back for IPv6?